web_src/js/features/repo-editor.js

```diff
@@ -120,18 +123,24 @@ export function initRepoEditor() {
         $(this)[0].setSelectionRange(0, 0);
       }
     }
-    parts = [];
-    $('.breadcrumb span.section').each(function () {
-      const element = $(this);
-      if (element.find('a').length) {
-        parts.push(element.find('a').text());
-      } else {
-        parts.push(element.text());
-      }
-    });
-    if ($(this).val()) parts.push($(this).val());
-    $('#tree_path').val(parts.join('/'));
-  }).trigger('keyup');
+
+    joinTreePath($(this));
+  });
+
+  $editFilename.on('keydown', function (e) {
+    const $section = $('.breadcrumb span.section');
+
+    if (e.keyCode === 8 && getCursorPosition($(this)) === 0 && $section.length > 0) {
```

Comment:
```suggestion
    // Jump back to last directory once the filename has been completely deleted
    if (e.code === 'Backspace' && getCursorPosition($(this)) === 0 && $section.length > 0) {
```
Or do I misunderstand this legacy code?

Reply:
Yes I think your understanding is corrent.

Reply:
> Or do I misunderstand this legacy code?

Applied the suggested changes, modified the comment a little bit though as "// Jump back to last directory once the filename is empty"